hbase-shell/src/main/ruby/hbase_constants.rb

```diff
@@ -109,8 +109,8 @@ def self.promote_constants(constants)
     end
   end
 
-  promote_constants(org.apache.hadoop.hbase.HColumnDescriptor.constants)
-  promote_constants(org.apache.hadoop.hbase.HTableDescriptor.constants)
+  promote_constants(org.apache.hadoop.hbase.client.ColumnFamilyDescriptorBuilder.constants)
```

Comment:
we'll need to release note the changed set of constants, if any.

Reply:
This patch removes 11 constants. I added these to the release notes and left a long-form explanation in Jira:

https://issues.apache.org/jira/browse/HBASE-20819?focusedCommentId=17145696

Reply:
Hurm. Some of those still look likely to be useful (e.g. MOB_THRESHOLD_BYTES). Can we add them in without bringing back references to the classes we're removing?

Reply:
I just pushed a commit that adds back three constants, but leaves the rest out. I left a [comment on Jira](https://issues.apache.org/jira/browse/HBASE-20819?focusedCommentId=17146470#comment-17146470) with my rationale for each of the constants. **Let me know if you disagree with any of my conclusions, and I can easily add more constants back** 😄.

Specifically for the [MOB constants](https://github.com/apache/hbase/blob/025ddce868eb06b4072b5152c5ffae5a01e7ae30/hbase-client/src/main/java/org/apache/hadoop/hbase/client/ColumnFamilyDescriptorBuilder.java#L163), we added setters for all the MOB key-value pairs to our builder. I think including these _private_ constants will encourage users to go against our defined interface.

Reply:
That's great work, thanks! Analysis looks reasonable to me.